internal/pkg/admission/policy-server-deployment_test.go

```diff
@@ -210,38 +211,247 @@ func TestIfPolicyServerImageChanged(t *testing.T) {
+		RunAsNonRoot:             &runAsRoot,
+		AllowPrivilegeEscalation: &allowPrivilegeEscalation,
+		Capabilities:             &capabilities,
+	}
+	policyServer := &policiesv1.PolicyServer{
+		Spec: policiesv1.PolicyServerSpec{
+			Image: "image",
+			SecurityContexts: policiesv1.PolicyServerSecurity{
+				Container: &containerSecurity,
+				Pod:       &podSecurity,
+			},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.ReadOnlyRootFilesystem != readOnlFileSystem {
+		t.Error("Policy server container ReadOnlyRootFilesystem diverge from the expected value")
+	}
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.Privileged != privileged {
+		t.Error("Policy server container Privileged diverge from the expected value")
+	}
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.RunAsNonRoot != runAsRoot {
+		t.Error("Policy server container RunAsNonRoot diverges from the expected value")
+	}
+	if *deployment.Spec.Template.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation != allowPrivilegeEscalation {
+		t.Error("Policy server container AllowPrivilegeEscalation diverge from the expected value")
+	}
+	if deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities == nil {
+		t.Error("Policy server container should have capabilities defined")
+	} else {
+		if len(deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Add) > 0 {
+			t.Error("Policy server container should not have 'Add' capabilities defined")
+		}
+		if len(deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Drop) != 1 ||
+			deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Drop[0] != dropCapabilityAll {
+			t.Error("Policy server container Capabilities should have only one 'All' drop capability")
+		}
+	}
+
+	if deployment.Spec.Template.Spec.SecurityContext == nil {
+		t.Error("Pod securityContext should be defined ")
+		return
+	}
+
+	if *deployment.Spec.Template.Spec.SecurityContext.RunAsUser != user {
+		t.Error("Pod RunAsUser diverges from the expected value")
+	}
+	if *deployment.Spec.Template.Spec.SecurityContext.RunAsGroup != group {
+		t.Error("Pod RunAsGroup diverges from the expected value")
+	}
+}
+
+func TestDefaultContainerSecurityContext(t *testing.T) {
+	securityContext := defaultContainerSecurityContext()
+
+	if *securityContext.ReadOnlyRootFilesystem != true {
+		t.Error("Policy server container ReadOnlyRootFilesystem diverge from the expected value")
+	}
+	if *securityContext.Privileged != false {
+		t.Error("Policy server container Privileged diverge from the expected value")
+	}
+	if *securityContext.RunAsNonRoot != true {
+		t.Error("Policy server container RunAsNonRoot diverges from the expected value")
+	}
+	if *securityContext.AllowPrivilegeEscalation != false {
+		t.Error("Policy server container AllowPrivilegeEscalation diverge from the expected value")
+	}
```

Comment:
Same here, we should use the value returned by the `defaultContainerSecurityContext` as expected values. Maybe we can even use some k8s library that does a deep comparison of the two objects

Reply:
Fine